### PR TITLE
[incubator/vault] Add support for creating a service account

### PR DESCRIPTION
**Pull request by jbialy**
#### What this PR does / why we need it:

In situations where we want Vault to have access to the api server a service account is required. An example for this is when using Vault's Kubernetes auth method. 

https://www.vaultproject.io/docs/auth/kubernetes.html#configuring-kubernetes

This PR enables service account creation if needed.

#### Checklist

- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by jbialy** [comment]
👀 PTAL!

**Comment by viglesiasce** [comment]
@jpds can you take a look?

**Comment by jbialy** [comment]
@jpds , when you have a minute. Can you PTAL?

**Comment by jbialy** [comment]
@mattfarina could you please take a look at this PR and approve it if it looks good? It'd be great to get it merged!

**Comment by qlikcoe** [comment]
@jbialy , There is a conflict because the chart version was bumped by other PRs...

**Comment by jbialy** [comment]
The tests are ✅. @viglesiasce @unguiculus @jpds . Can you guys give this a `/lgtm` ?

**Comment by jbialy** [comment]
Bump! PTAL!

**Review by mattfarina — APPROVED** [review]

File: incubator/vault/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "vault.serviceAccountName" . }}
+  labels:
+    app: {{ template "vault.name" . }}
+    chart: {{ template "vault.chart" . }}
```
Comment (therealgambo) [inline incubator/vault/templates/serviceaccount.yaml:8]: This is incorrect as the `vault.chart` template doesn't exist - did you forget to add it?

Or you can use what the other vault manifests use: `{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}`

Any chance of a quick fix for this @jbialy ?

**Review by therealgambo — COMMENTED** [review]
This line is causing a templating error because the template function doesn't exist.

File: incubator/vault/templates/serviceaccount.yaml (hunk 1 above)
Comment (jbialy) [reply incubator/vault/templates/serviceaccount.yaml:8]: Ah, good catch! I likely grabbed the `serviceaccount` definition from another chart, and overlooked the way the `chart` label is set!

I'll adjust it to use `{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}` as you suggested!

**Pull request merged**